src/lib/src/paginator/paginator.component.ts

```diff
@@ -53,11 +56,16 @@ import { TsPaginationMenuItem } from './../utilities/interfaces';
  * <example-url>https://goo.gl/ieUPaG</example-url>
  */
 @Component({
-  selector: 'ts-pagination',
-  templateUrl: './pagination.component.html',
-  styleUrls: ['./pagination.component.scss'],
+  selector: 'ts-paginator',
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss'],
+  host: {
+    class: 'ts-paginator',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
Why is here `.OnPush` - CheckOnce not CheckAlways?

Reply:
This basically tells Angular to only run change detection when an `@Input` value changes. You are in essence promising to not do anything to mutate the state internally. This is much more performant since the change detection cycle doesn't have to check everything each time.

The few times where we need to do something internally in a component that uses this `changeDetection` strategy we manually tell the change detector to check for changes using `changeDetectorRef.markForCheck()`.